File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int k=1;
    public int height(TreeNode root){
        if(root == null) return 0;
        int left = height(root.left);
        int right = height(root.right);
        if(Math.abs(left-right) >1){
            k = 0;
        }
        return Math.max(left,right)+1;
    }
    public boolean isBalanced(TreeNode root) {
        height(root);
        if(k==0) return false;
        return true;
    }
}

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int dfs(TreeNode* root, bool& flag){
        if(!root) return 0;
        int lans= dfs(root->left, flag);
        int rans= dfs(root->right, flag);
        if(abs(lans-rans)>1) flag= false;
        return max(lans,rans)+1;
    }
    bool isBalanced(TreeNode* root) {
        bool flag= true;
        dfs(root, flag);
        return flag;
    }
};